django-stubs/http/request.pyi

```diff
@@ -51,7 +52,7 @@ class HttpRequest(BytesIO):
     resolver_match: ResolverMatch = ...
     content_type: Optional[str] = ...
     content_params: Optional[Dict[str, str]] = ...
-    user: AbstractBaseUser
+    user: Union[AbstractBaseUser, AnonymousUser]
```

Comment:
Let's test it! In the best world `is_authenticated` function should resolve this type.

Reply:
I'm not sure how to test it properly yet (unfortunately I'm not so good at testing) so please label it with "help wanted", maybe someone will be able to do this, or I will get back to this later with more experience.

Reply:
You can have a look at https://github.com/typeddjango/django-stubs/blob/master/test-data/typecheck/models/test_contrib_models.yml and https://github.com/typeddjango/django-stubs/blob/master/test-data/typecheck/contrib/auth/test_decorators.yml

Reply:
I've tested the tags, but still have no idea how to make `is_authenticated` resolve `Union[]` type. Any further tips?